test/jdk/javax/swing/JTabbedPane/bug4666224.java

```diff
@@ -0,0 +1,168 @@
+                        and repeat from step 1 to 5.
+                    7. Type 'R' to align the tabs to the right side and repeat
+                        from step 1 to 5.
+                    8. Type 'B' to align the tabs to the bottom side and repeat
+                        from step 1 to 5.
+                    9. Type 'L' to align the tabs to the left side and repeat
+                        from step 1 to 5.
+                    10. Type 'T' to align the tabs to the top side and repeat
+                         from step 1 to 5.""";
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("JTabbedPane Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(20)
+                .columns(40)
+                .testUI(bug4666224::createAndShowUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static JFrame createAndShowUI() {
+        frame = new JFrame("Test JTabbedPane");
+        tabPane = new JTabbedPane();
+        tabPane.addMouseListener(new MouseAdapter() {
+            public void mouseClicked(MouseEvent e) {
+                Point pt = e.getPoint();
+                System.out.println("Index at location: "
+                        + tabPane.indexAtLocation(pt.x, pt.y));
+            }
+        });
+        InputMap inputMap = createInputMap();
+        SwingUtilities.replaceUIInputMap(frame.getRootPane(), JComponent.WHEN_IN_FOCUSED_WINDOW, inputMap);
+        ActionMap actionMap = createActionMap();
+        SwingUtilities.replaceUIActionMap(frame.getRootPane(), actionMap);
+
+        tabPane.setTabPlacement(JTabbedPane.TOP);
+        tabPane.setTabLayoutPolicy(JTabbedPane.SCROLL_TAB_LAYOUT);
+
+        for (int i = 0; i <= 10; i++) {
+            JPanel panel = new JPanel();
+            panel.setPreferredSize(new Dimension(500, 200));
+            tabPane.addTab("Tab " + i, panel);
+        }
+        JPanel mainPanel = new JPanel();
+        mainPanel.add(tabPane);
+
+        frame.setSize(600, 300);
+        frame.getContentPane().add(mainPanel);
+        tabPane.requestFocus();
+        return frame;
+    }
+
+    protected static InputMap createInputMap() {
+        return LookAndFeel.makeComponentInputMap(frame.getRootPane(), new Object[] {
+                "R", "right",
+                "L", "left",
+                "T", "top",
+                "B", "bottom",
+                "C", "changeLayout",
+                "D", "dump"
+        });
+    }
+
+    protected static ActionMap createActionMap() {
+        ActionMap map = new ActionMap();
+        map.put("right", new RotateAction(JTabbedPane.RIGHT));
+        map.put("left", new RotateAction(JTabbedPane.LEFT));
+        map.put("top", new RotateAction(JTabbedPane.TOP));
+        map.put("bottom", new RotateAction(JTabbedPane.BOTTOM));
+        map.put("changeLayout", new ChangeLayoutAction());
+        map.put("dump", new DumpAction());
```

Comment:
`DumpAction `can also be mentioned in INSTRUCTIONS I guess, since its been used here.

Reply:
There is no instruction mentioned in old applet test. Kept it as it is.

Reply:
I guess you can update it since its there in the test, still not mandatory.